src/component/Task/Task.js

```diff
@@ -0,0 +1,318 @@
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid'; // 고유 ID 생성을 위해 uuid 사용
+import styled from 'styled-components';
+import { getTodoProgressForDate, saveTasksToLocalStorage } from '../../utils/LocalStorageUtil';
+import { CalendarDays, CopyPlus, Medal, Plus, Undo2 } from 'lucide-react';
+import PendingTask from './components/PendingTask';
+import CompletedTask from './components/CompletedTask';
+import TabIndicator from './components/TabIndicator';
+import Tooltip from '../../utils/Tooltip';
+import Swal from 'sweetalert2';
+import { getDayName } from '../../utils/dateUtil';
+
+
+
+function Task({ selectedDate, handleCloseModal }) {
+    const [tasks, setTasks] = useState(new Map()); // 해시맵 구조로 tasks를 관리
+    const [newTask, setNewTask] = useState('');
+    const [addModal, setAddModal] = useState(false);
+    const [activeIndex, setActiveIndex] = useState(0);
+    const [tooltip, setTooltip] = useState(null); // tooltip 상태
+    const { completed, total } = getTodoProgressForDate(selectedDate); // 성취도 계산
+    const formattedDate = selectedDate.toISOString().split('T')[0];
+
+    const updateTasks = (newTasks) => {
+        setTasks(new Map(newTasks)); // 중복 제거
+        saveTasksToLocalStorage(formattedDate, Array.from(newTasks.values()));
+    };
+    
+    
+    const handleAddTask = () => {
+        if (newTask.trim() === '') return;
+        const taskId = uuidv4();
+        const createdTime = new window.Date().toLocaleTimeString(); 
+    
+        const newTaskObj = { 
+          id: taskId, 
+          text: newTask, 
+          completed: false, 
+          completedTime: null,
+          createdTime: createdTime // 작성 시간 추가
+        };
+    
+        // 기존 Map 수정 후 상태 업데이트
+        tasks.set(taskId, newTaskObj);
+        setNewTask('');
+        updateTasks(tasks);
+    };
```

Comment:
```suggestion
//상단
const [task, setTask] =useState({}) 

...
       setTasks(prevTasks => ({
       ...prevTasks,
       [taskId] : newTaskObj
       })));
     setNewTask('')
     ...
    };
```

tasks 상태를 직접 변경하는 것보다 새로운 객체나 배열을 생성해서 상태를 업데이트하는 게 오류 방지에 더 좋을 것 같습니다!

Reply:
코드 가독성 측면에서 좀 줄여보자는 생각으로 구현했었는데 밑에 알려주신 링크와 조금 더 알아보니 새로 생성해서 업데이트하는 게 좋을 것 같습니다! 가독성, 에러방지, 효율 등에서 적절한 선을 찾아야겠네요 ㅎㅎ 감사합니다!